hieroplatform/src/test/java/org/hiero/sketch/HistogramTest.java

```diff
@@ -0,0 +1,188 @@
+        Histogram1D hist1 = new Histogram1D(buckDes);
+        DoubleArrayColumn col1 = DoubleArrayTest.generateDoubleArray(20000);
+        FullMembership fmap1 = new FullMembership(20000);
+        hist1.createHistogram(col1, fmap1, null);
+        size = 0;
+        for (int i = 0; i < bucketnum; i++)
+            size += hist1.getBucket(i).getCount();
+        assertEquals(size + hist1.getMissingData() + hist1.getOutOfRange(), 20000);
+        Histogram1D hist2 = hist1.union(hist);
+        size = 0;
+        for (int i = 0; i < bucketnum; i++)
+            size += hist2.getBucket(i).getCount();
+        assertEquals(size + hist2.getMissingData() + hist2.getOutOfRange(), 30000);
+        Histogram1D hist3 = new Histogram1D(buckDes);
+        DoubleArrayColumn col2 = DoubleArrayTest.generateDoubleArray(20000);
+        FullMembership fmap2 = new FullMembership(20000);
+        hist3.createSampleHistogram(col2, fmap2, null, 0.1);
+        size = 0;
+        for (int i = 0; i < bucketnum; i++)
+            size += hist3.getBucket(i).getCount();
+        assertEquals(size + hist3.getMissingData() + hist3.getOutOfRange(), 2000);
+        }
+
+    @Test
+    public void testHistogram1DLight() throws Exception {
+        final int bucketnum = 110;
+        BucketsDescriptionEqSize buckDes = new BucketsDescriptionEqSize(0,100,bucketnum);
+        Histogram1DLight hist = new Histogram1DLight(buckDes);
+        DoubleArrayColumn col = DoubleArrayTest.generateDoubleArray(10000);
+        FullMembership fmap = new FullMembership(10000);
+        hist.createHistogram(col, fmap, null);
+        int size = 0;
+        for (int i = 0; i < bucketnum; i++)
+            size += hist.getBucket(i);
+        assertEquals(size + hist.getMissingData() + hist.getOutOfRange(), 10000);
+        Histogram1DLight hist1 = new Histogram1DLight(buckDes);
+        DoubleArrayColumn col1 = DoubleArrayTest.generateDoubleArray(20000);
+        FullMembership fmap1 = new FullMembership(20000);
+        hist1.createHistogram(col1, fmap1, null);
+        size = 0;
+        for (int i = 0; i < bucketnum; i++)
+            size += hist1.getBucket(i);
+        assertEquals(size + hist1.getMissingData() + hist1.getOutOfRange(), 20000);
+        Histogram1DLight hist2 = hist1.union(hist);
+        size = 0;
+        for (int i = 0; i < bucketnum; i++)
+            size += hist2.getBucket(i);
+        assertEquals(size + hist2.getMissingData() + hist2.getOutOfRange(), 30000);
+        Histogram1DLight hist3 = new Histogram1DLight(buckDes);
+        DoubleArrayColumn col2 = DoubleArrayTest.generateDoubleArray(20000);
+        FullMembership fmap2 = new FullMembership(20000);
+        hist3.createSampleHistogram(col2, fmap2, null, 0.1);
+        size = 0;
+        for (int i = 0; i < bucketnum; i++)
+            size += hist3.getBucket(i);
+        assertEquals(size + hist3.getMissingData() + hist3.getOutOfRange(), 2000);
```

Comment:
This is surprising, you always get exactly 2000 items?
I hope you are not paying too much for that; we seldom - if ever - will need this degree of precision.